server/bin/pbench-backup-tarballs

```diff
@@ -1,83 +1,123 @@
+
+# Correct the first path to be the "lib" directory to make the
+# pbench-backup-to-s3 script simple, allowing the imports from the
+# backups3 library to work cleanly.
+first_path = sys.path[0]
+assert first_path.endswith('/bin')
+lib_path = os.path.join(os.path.dirname(first_path), 'lib')
+sys.path[0] = lib_path
+
+from pbench import PbenchConfig, BadConfig, get_pbench_logger
+from s3backup import S3Config
+
+_NAME_    = "pbench-backup-tarballs"
+
+# Global logger for the module, setup in main()
+_logger = None
+
+def rsync_archive_backup(config, report):
+    # Function to do rsync between archive and backup
+    if not os.path.isdir(config.ARCHIVE):
+        _logger.error('Archive: {} not found.'.format(config.ARCHIVE))
+        os._exit(1)
+    if not os.path.isdir(config.BACKUP):
+        _logger.error('Backup: {} not found.'.format(config.BACKUP))
+        os._exit(1)
+    exclude = ["--exclude={}".format(i) for i in config.EXCLUDE_DIRS.split(" ")]
     # N.B The trailing slash is important
-    rsync -va --stats $exclude $rla/ $rlb | tee -a $stats
-    rsync_sts=$?
-    if [[ $rsync_sts -ne 0 ]]; then
-        echo "$PROG: rsync failed with code: $rsync_sts, $rlb"
-    fi
-fi
-
-echo "end-$(timestamp)"
-
-log_finish
-
-# Back up some way and find the empty line separating the list of files
-# from the stats.
-cat << EOF > $index_content
-$PROG.$TS($PBENCH_ENV)
-EOF
-
-if [ -e $stats ] ;then
-    tail -n 20 $stats | sed -n '/^$/,$p' >> $index_content
-fi
-
-pbench-report-status --name $PROG --timestamp $TS --type status $index_content
-
-exit 0
+    archive = "{}/".format(config.ARCHIVE)
+    stat_out = sh.rsync("-va", "--stats", exclude, archive, config.BACKUP)
+    if stat_out.exit_code == 0:
+        _logger.info(str(stat_out))
+    elif stat_out.exit_code != 0:
+        _logger.warning("Failed to do rsync between {} and {}: {}\n".format(config.ARCHIVE, config.BACKUP, e))
+        report.write("Failed to do rsync between {} and {}: {}\n".format(
+            config.ARCHIVE, config.BACKUP, e))
+
+def backup_data_to_s3(s3_obj, config, report):
+    # Function to copy data to s3
+    try:
+        tarlist = glob.glob('{}/*/*.tar.xz'.format(config.ARCHIVE))
+        s3_obj.upload_to_bucket(tarlist, report, _logger)
+    except Exception as e:
+        _logger.warning("Failed to upload data from {} to s3: {}\n".format(config.ARCHIVE, e))
+        report.write("Failed to upload data from {} to s3: {}\n".format(config.ARCHIVE, e))
+
+def main(options):
+    if not options.cfg_name:
+        print("{}: ERROR: No config file specified; set CONFIG env variable or"
+                " use --config <file> on the command line".format(_NAME_),
+                file=sys.stderr)
+        return 2
+
+    try:
+        config = PbenchConfig(options.cfg_name)
+    except BadConfig as e:
+        print("{}: {}".format(_NAME_, e), file=sys.stderr)
+        return 3
+
```

Comment:
PbenchConfig() checks a few required things and bails if they are not there, but it also sets a bunch without checking that the thing exists, e.g. it sets config.ARCHIVE to the directory that the config file says, but it does not check whether the directory exists. Since both this script and the verify script cannot do anything if the archive dir does not exist, you should check that (both here and in the verify script) and bail if necessary.